src/main/java/streams/cta/io/CSVWriter.java

```diff
@@ -0,0 +1,74 @@
+package streams.cta.io;
+
+import com.google.common.base.Joiner;
+import stream.Data;
+import stream.Keys;
+import stream.ProcessContext;
+import stream.StatefulProcessor;
+import stream.annotations.Parameter;
+import stream.io.SourceURL;
+
+import java.io.FileOutputStream;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * CSVWriter writes out the values for the given keys to a csv file.
+ * The first line of the CSV file is a header.
+ * The header contains the names of the keys found in the data item.
+ *
+ * @author kai
+ */
+public class CSVWriter implements StatefulProcessor {
```

Comment:
CSVWriter might be used by more than one computing node at the time. Are we interested in parallelizing the write process between multiple nodes? (e.g. writing in temp files and then just appending them to a single global output file which is then done in a single mode).